### PR TITLE
Downgrade PHP version to 7.0

### PR DESCRIPTION
**Pull request by monobook**
| Q               | A
| --------------- | ---
| Bug fix?        | no|yes
| New feature?    | no|yes
| BC breaks?      | no|yes
| Deprecations?   | no|yes
| License         | MIT

#### What's in this PR?

Downgrade composer PHP version to 7.0

#### Why?

Because we have old versions of PHP and Symfony as well in the world.

### CODE REVIEW

**Comment by monobook** [comment]
@dbu ping :)

File: composer.json [hunk 1]
```diff
@@ -11,16 +11,16 @@
...
-        "guzzlehttp/psr7": "^1.4",
-        "php-http/client-common": "^2.0",
-        "psr/log": "^1.0",
-        "symfony/filesystem": "^3.4|^4.0",
-        "symfony/options-resolver": "^3.4|^4.0"
+        "php": "^7.0",
+        "php-http/client-common": "^1.9|^2.0",
+        "guzzlehttp/psr7": "^1.5",
+        "psr/log": "^1.1",
+        "symfony/filesystem": "^2.8|^3.4|^4.0",
```
Comment (xabbuh) [inline composer.json:18]: IMO adding support for Symfony versions that are not supported anymore doesn't make much sense.

Comment (monobook) [reply composer.json:18]: okay, but what about projects with 2.8 and currently not planned (or never planned) upgrade to 3.*?

Comment (monobook) [reply composer.json:18]: main point is to have a branch or tag with work functional for all versions. or use separate tag or branch (for example v1 or v2) for "new" components or modern versions...

Comment (xabbuh) [reply composer.json:18]: IMO if one decides to stay on unsupported framework versions, one also needs to accept that they cannot use new features.

Comment (monobook) [reply composer.json:18]: okay

**Comment by sagikazarmark** [comment]
I'm sorry, but I'm not sure we want this. PHP 7.3 is the latest stable, so if we follow the common practice of supporting the last two versions, we should upgrade, not downgrade.

I'll let others join the conversation before making a decision, but I'm quite sure they'll agree with me.

**Review by dbu — CHANGES_REQUESTED** [review]
i agree with xabbuh and mark that we do not want to downgrade this code. php 7.0 is not maintained anymore, and symfony 2.8 is really old. the changes are also a BC break because the interface signature changes.

if you are stuck on a legacy project where more testing is to be added but no symfony upgrade is done, you can of course use your fork.

**Comment by monobook** [comment]
> i agree with xabbuh and mark that we do not want to downgrade this code. php 7.0 is not maintained anymore, and symfony 2.8 is really old. the changes are also a BC break because the interface signature changes.
> 
> if you are stuck on a legacy project where more testing is to be added but no symfony upgrade is done, you can of course use your fork.

ok lets use fork :) tnx

**Comment by dbu** [comment]
thanks for your understanding :+1:

**Pull request closed without merging**